src/Workspaces/Core/Portable/Workspace/Solution/Checksum_Factory.cs

```diff
@@ -14,13 +14,50 @@ namespace Microsoft.CodeAnalysis
     // all these are just helper methods
     internal partial class Checksum
     {
+        private static readonly Stack<IncrementalHash> s_hashStack = new Stack<IncrementalHash>();
```

Comment:
Looking at the impl, it looks like Dispose ends up calling this:

https://github.com/dotnet/corefx/blob/bffef76f6af208e2042a2f27bc081ee908bb390b/src/Common/src/Internal/Cryptography/HashProviderCng.cs#L131

So if we leak, we'll end up not properly getting rid of bcrypt handles.